restli-tools/src/main/java/com/linkedin/restli/tools/clientgen/FluentApiGenerator.java

```diff
@@ -0,0 +1,257 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.apache.velocity.runtime.log.Log4JLogChute;
+import org.apache.velocity.runtime.resource.loader.JarResourceLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Generate fluent api client bindings from idl file to java source file.
+ *
+ * @author Karthik Balasubramanian
+ */
+public class FluentApiGenerator
+{
+  private static final Logger LOGGER = LoggerFactory.getLogger(FluentApiGenerator.class);
+  private static final Options OPTIONS = new Options();
+  private static final Map<ResourceMethod, String> BUILDER_BASE_MAP = new HashMap<>();
+  private static final String API_TEMPLATE_DIR = "apiVmTemplates";
+
+  static {
+    BUILDER_BASE_MAP.put(ResourceMethod.GET, "get");
+    BUILDER_BASE_MAP.put(ResourceMethod.DELETE, "delete");
+    BUILDER_BASE_MAP.put(ResourceMethod.UPDATE, "update");
+    BUILDER_BASE_MAP.put(ResourceMethod.CREATE, "create");
+    BUILDER_BASE_MAP.put(ResourceMethod.PARTIAL_UPDATE, "partialUpdate");
+    BUILDER_BASE_MAP.put(ResourceMethod.GET_ALL, "getAll");
+    BUILDER_BASE_MAP.put(ResourceMethod.BATCH_GET, "batchGet");
+    BUILDER_BASE_MAP.put(ResourceMethod.BATCH_UPDATE, "batchUpdate");
+    BUILDER_BASE_MAP.put(ResourceMethod.BATCH_PARTIAL_UPDATE, "batchPartialUpdate");
+    BUILDER_BASE_MAP.put(ResourceMethod.BATCH_DELETE, "batchDelete");
+    BUILDER_BASE_MAP.put(ResourceMethod.BATCH_CREATE, "batchCreate");
+  }
+
+  public static void main(String[] args) throws Exception
+  {
+    OPTIONS.addOption("h", "help", false, "Show help.");
+    OptionBuilder.withLongOpt("targetDir");
+    OptionBuilder.hasArgs(1);
+    OptionBuilder.isRequired();
+    OptionBuilder.withDescription("Target directory in which the classes should be generated.");
+    OPTIONS.addOption(OptionBuilder.create('t'));
+    OptionBuilder.withLongOpt("resolverPath");
+    OptionBuilder.hasArgs(1);
+    OptionBuilder.isRequired();
+    OptionBuilder.withDescription("Resolver path for loading data schemas");
+    OPTIONS.addOption(OptionBuilder.create('p'));
+    OPTIONS.addOption("r", "rootPath", true, "Root path to use for generating relative path for source location");
+
+    try
+    {
+      final CommandLineParser parser = new GnuParser();
+      CommandLine cl = parser.parse(OPTIONS, args);
+
+      if (cl.hasOption('h'))
+      {
+        help();
+        System.exit(0);
+      }
+      String targetDirectory = cl.getOptionValue('t');
+      String resolverPath = cl.getOptionValue('p');
+      if (ArgumentFileProcessor.isArgFile(resolverPath))
+      {
+        resolverPath = ArgumentFileProcessor.getContentsAsArray(resolverPath)[0];
+      }
+      String[] sources = cl.getArgs();
+      if (sources.length == 1 && ArgumentFileProcessor.isArgFile(sources[0]))
```

Comment:
though I know ArgFile is quite common now and a bit internal, I actually hope this comment can be surfaced somewhere, through help doc or something

Reply:
Good point. Updated to include help on using arg file.